packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/UiThreadUtil.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+ package com.facebook.react.bridge
+
+ import android.os.Handler
+ import android.os.Looper
+ import com.facebook.react.common.build.ReactBuildConfig
+ 
+ /**
+  * Utility for interacting with the UI thread.
+  */
+ public object UiThreadUtil {
+ 
+     @Volatile private var mainHandlerInternal: Handler? = null
+ 
+     /**
+      * Handler associated with the main (UI) thread.
+      * Exposed for Java interop as `getUiThreadHandler()`.
+      */
+     private val mainHandler: Handler
+         get() {
+             if (mainHandlerInternal == null) {
+                 synchronized(this) {
+                     if (mainHandlerInternal == null) {
+                         mainHandlerInternal = Handler(Looper.getMainLooper())
+                     }
+                 }
+             }
+             return mainHandlerInternal!!
+         }
+ 
+     /**
+     * Java-compatible static getter for the UI thread handler.
+     */
+     @JvmStatic
+     public fun getUiThreadHandler(): Handler = mainHandler
+     /**
+      * Checks if the current thread is the UI thread.
+      *
+      * @return `true` if the current thread is the UI thread, `false` otherwise
+      */
+     @JvmStatic
+     public fun isOnUiThread(): Boolean = Looper.getMainLooper().thread == Thread.currentThread()
+ 
+     /**
+      * Throws an [AssertionException] if the current thread is not the UI thread.
+      * This method is only active in debug mode and is a no-op in production.
+      */
+     @JvmStatic
+     public fun assertOnUiThread() {
+         if (ReactBuildConfig.DEBUG) {
+             SoftAssertions.assertCondition(isOnUiThread(), "Expected to run on UI thread!")
+         }
+     }
+ 
+     /**
+      * Throws an [AssertionException] if the current thread is the UI thread.
+      * This method is only active in debug mode and is a no-op in production.
+      */
```

Comment:
The comments seem slighly different in terms of content as how they were originally in Java, can we try keeping them as close as how they were? feel free to keep the extra improvements and grammar corrections 

```suggestion
 /**
   * Throws an [AssertionException] if the current thread is the UI thread. This is a noop in
   * production, and is only meant to run in debug mode! If you need to check for incorrect-thread
   * issues in production, duplicate this code and call it elsewhere.
   */
```

Reply:
Done